mpegts/mpegts_demuxer.cpp

```diff
@@ -97,85 +97,87 @@ uint8_t MpegTsDemuxer::decode(SimpleBuffer &rIn) {
 
             if (lTsHeader.mAdaptationFieldControl == MpegTsAdaptationFieldType::mPayloadOnly ||
                 lTsHeader.mAdaptationFieldControl == MpegTsAdaptationFieldType::mPayloadAdaptionBoth) {
-                PESHeader lPesHeader;
-                if (lTsHeader.mPayloadUnitStartIndicator == 0x01) {
 
-                    mEsFrames[lTsHeader.mPid]->mRandomAccess = lRandomAccessIndicator;
+                EsFrame* lEsFrame = mEsFrames[lTsHeader.mPid].get();

```

Comment:
Do you really need the `get()` here? Why don't get a shared pointer or a reference to the shared pointer?